### PR TITLE
Improved reconnection delay

### PR DESCRIPTION
**Pull request by albaintor**
I disabled the setup call during (re)connection when it has already been done once.
I also updated the requirements (denonavr library in particular)

### CODE REVIEW

**Comment by albaintor** [comment]
I also updated pyee requirement but I realized maybe I shouldn't as you mentioned limits with pylint

**Comment by albaintor** [comment]
Update : I have tested the fix on my setup and it working. I was worried that the setup call was always necessary after disconnect but this is not the case. I have mocked a script to test : 1/ connect, 2/disconnect, 3/connect again, and I still have the events (volume change, input source changed....)

File: intg-denonavr/avr.py [hunk 1]
```diff
@@ -800,29 +819,16 @@ async def setup(self) -> ucapi.StatusCodes:
...
             await self.setup_open()
 
-    @async_handle_denonlib_errors
-    async def send_command(self, cmd: str) -> ucapi.StatusCodes:
-        """Send a command to the AVR."""
-        if self._use_telnet:
-            await self._receiver.async_send_telnet_commands(cmd)
-        else:
-            url = AVR_COMMAND_URL + "?" + cmd.replace(" ", "%20")
-            await self._receiver.async_get_command(url)
```
Comment (zehnm) [inline intg-denonavr/avr.py:810]: What was the reason to remove this method and duplicate it 10 times?

File: intg-denonavr/avr.py [hunk 2]
```diff
@@ -800,29 +819,16 @@ async def setup(self) -> ucapi.StatusCodes:
...
-
     def _increase_expected_volume(self):
         """Without telnet, increase expected volume and send update event."""
         if not self._use_telnet or self._expected_volume is None:
             return
-        self._expected_volume = min(self._expected_volume + self._volume_step, 100)
-        # Send updated volume if no update task in progress
-        if not self._update_lock.locked():
-            self._event_loop.create_task(self._receiver.async_update())
+        self._expected_volume = min(self._expected_volume + VOLUME_STEP, 100)
```
Comment (zehnm) [inline intg-denonavr/avr.py:826]: Doesn't this ignore now the user configurable volume step?
- `self._volume_step` should contain the configured step from the setup flow
- `VOLUME_STEP` is a constant

File: intg-denonavr/avr.py [hunk 3]
```diff
@@ -800,29 +819,16 @@ async def setup(self) -> ucapi.StatusCodes:
...
     def _decrease_expected_volume(self):
         """Without telnet, decrease expected volume and send update event."""
         if not self._use_telnet or self._expected_volume is None:
             return
-        self._expected_volume = max(self._expected_volume - self._volume_step, 0)
-        # Send updated volume if no update task in progress
-        if not self._update_lock.locked():
-            self._event_loop.create_task(self._receiver.async_update())
+        self._expected_volume = max(self._expected_volume - VOLUME_STEP, 0)
+        self.events.emit(Events.UPDATE, self.id, {MediaAttr.VOLUME: self._expected_volume})
```
Comment (zehnm) [inline intg-denonavr/avr.py:834]: Please explain the volume logic change and removing the `_update_lock`.
This PR is about the reconnection delay. If that's not related, then please open a separate PR.

File: intg-denonavr/avr.py [hunk 4]
```diff
@@ -652,13 +650,6 @@ async def power_off(self) -> ucapi.StatusCodes:
         if not self._use_telnet:
             self._set_expected_state(States.OFF)
 
-    @async_handle_denonlib_errors
-    async def power_toggle(self) -> ucapi.StatusCodes:
-        """Send power-on or -off command to AVR based on current power state."""
-        if self._receiver.power is not None and self._receiver.power == "ON":
-            await self.power_off()
-        await self.power_on()
-
```
Comment (zehnm) [inline intg-denonavr/avr.py:661]: This method is still used!
https://github.com/unfoldedcircle/integration-denonavr/blob/main/intg-denonavr/media_player.py#L247-L248

**Review by zehnm — CHANGES_REQUESTED** [review]
Thanks for the PR. I only did a code review so far and I'll test the reconnection behaviour tomorrow.

There are a lot of changes which don't seem related to this PR's description.
I thought this is mainly to avoid calling `async_setup()` after a reconnect, but the other changes are not explained why they are required. For example removing the lock or changing volume handling.
Also the `power_toggle` method is still used and may not be removed.

**Comment by albaintor** [comment]
Sorry I messed up with the merge I committed experimental changes. This is reverted now

**Comment by zehnm** [comment]
Thanks for reverting the other changes. I'll start testing it tomorrow and look into the `receiver.async_setup()` function, if there could be any side-effects if not called at reconnect.

**Comment by albaintor** [comment]
Normally there shouldn't be: I have mocked and tested the reconnection but a second look is useful

**Comment by zehnm** [comment]
Looking at the code of the denonavr library, it seems safe to skip the `async_setup` call during reconnect and only call `async_update`. If setup is required, it will automatically be called:

https://github.com/ol-iver/denonavr/blob/72294567cd35904e7b5ee68595f577eea603ce66/denonavr/denonavr.py#L137-L169

I still need to test it with my receiver.

**Review by zehnm — APPROVED** [review]
LGTM 👍

**Pull request merged**